### PR TITLE
Cleaning lints

### PR DESCRIPTION
**Pull request by lukaslueg**
I've started to clean some minor defects in cargo. This is the first commit of possibly many.

Requesting advice if this is actually wanted; #cargo was positive.

Some things raise the minimum version of rust required to compile cargo. E.g. `assert_ne!(foo, bar)` instead of `assert!(foo != bar)` requires (iirc) rust 1.13. Any advice on that in particular?

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Comment by lukaslueg** [comment]
@alexcrichton, I understand that implicitly raising the minimal rust version required to compile cargo is a non-problem?

**Comment by alexcrichton** [comment]
Indeed!

File: src/cargo/util/config.rs [hunk 1]
```diff
@@ -386,7 +386,7 @@ impl Config {
         let cfg_verbose = self.get_bool("term.verbose").unwrap_or(None).map(|v| v.val);
         let cfg_color = self.get_string("term.color").unwrap_or(None).map(|v| v.val);
 
-        let color = color.as_ref().or(cfg_color.as_ref());
+        let color = color.as_ref().or_else(|| cfg_color.as_ref());
```
Comment (tamird) [inline src/cargo/util/config.rs]: there are a bunch of `.foo(|| someFunc())` in this PR - can any of them be rewritten by passing the function directly: `.foo(someFunc)`? In this case, that would be `color.as_ref().or_else(cfg_color.as_ref)`.

Note that in this specific case something even better is possible: `color.or(cfg_color).as_ref()`

Comment (lukaslueg) [reply src/cargo/util/config.rs]: You are right, that was an oversight. I have many more changes waiting and will incorporate this in the next PR.

**Pull request merged**